### DISCUSSION TITLE
Table positioning in fpdf2

### DISCUSSION THREAD

**Discussion by Constance2023** [py-pdf/fpdf2#918] [Q&A] [answered]
I am trying to 1) position a **table** below another cell. Then 2) position another **table** right of that.

When I align **table** 'left', and align the other **table** 'right', the second **table** ('right') is positioned below the first (on the right side of the page), instead of next to it.

I am looking for a way to position two **tables** at a precise location on the page, just as can be done with text.

Any ideas or examples how this can be solved?

**Comment by gmischler** [comment]
Welcome to fpdf2, @Constance2023,

like most content in fpdf2, tables are placed between the page margins horizontally (depending on the "align" parameter), and below the current y position vertically.

So if you want to use left and right alignment for your horizontal placement, then all you need to do is to remember the current value of `pdf.y` before starting the first table, and reset it to that value before starting the second one.

**Reply by Constance2023** [reply]
Thanks for your quick  reply

My example code

```python
with pdf.table(borders_layout="SINGLE_TOP_LINE",
               cell_fill_color=200,  # grey
               cell_fill_mode="ROWS",
               line_height=pdf.font_size * 1.1,
               text_align=("LEFT", "RIGHT"),
               width=80,
               align="LEFT") as table:
    for data_row in data:
        row = table.row()
        for datum in data_row:
            row.cell(datum)
            
with pdf.table(borders_layout="SINGLE_TOP_LINE",
               cell_fill_color=200,  # grey
               cell_fill_mode="ROWS",
               line_height=pdf.font_size * 1.1,
               text_align=("LEFT", "RIGHT"),
               width=80,
               align="RIGHT") as table:
    for data_row in data:
        row = table.row()
        for datum in data_row:
            row.cell(datum)
```

1) Where in the above do I see or place 'current value of pdf.y'
2) Where do I reset it

With Cell, Multi-Cell and Text it's all straight forward - for table I see no docs or examples that show where to manipulate or place the alignment with coordinates or after cell, multi cell etc...

[gmischler: fixed code formatting]

**Reply by gmischler** [reply, marked as answer]
Before starting the first table, eg.:

`table_top_y = pdf.y`

and before starting the second table:

`pdf.y = table_top_y`

the "y" attribute of the FPDF class is how it maintains its current vertical position internally. You can check its value at any time. You can obviously also change it, if you want to deviate from the standard top-down sequence of element placement on the page.
